src/flow/mincostflow.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <bits/stdc++.h>
+using namespace std;
+
+// 最小費用流 O(FElogV)
+struct MinCostFlow {
+    int V;
+    vector<vector<vector<long long>>> g;  // g[from] = {{to, 容量, コスト, 逆辺のindex} ... }
+    vector<long long> h, dis;             // ポテンシャル, 最短距離
+    vector<int> prevv, preve;             // 直前の頂点, 辺
+
+    MinCostFlow(int v) : V(v), g(v), dis(v), prevv(v), preve(v) {
+    }
+
+    void add_edge(int u, int v, long long c, long long d) {
+        g[u].push_back({v, c, d, (int)g[v].size()});
+        g[v].push_back({u, 0, -d, (int)g[u].size() - 1});
+    }
+
+    long long min_cost_flow(int s, int t, long long f) {
+        long long res = 0;
+        h.assign(V, 0);
+        while (f > 0) {
+            priority_queue<vector<long long>, vector<vector<long long>>, greater<vector<long long>>> que;
+            dis.assign(V, LLONG_MAX);
+            dis[s] = 0;
+            que.push({0, s});
+            while (que.size()) {
+                int d = que.top()[0];
+                int v = que.top()[1];
+                que.pop();
+                if (dis[v] < d) continue;
+                for (int i = 0; i < g[v].size(); i++) {
+                    auto now = g[v][i];
+                    if (now[1] > 0 and dis[now[0]] > dis[v] + now[2] + h[v] - h[now[0]]) {
+                        dis[now[0]] = dis[v] + now[2] + h[v] - h[now[0]];
+                        prevv[now[0]] = v;
+                        preve[now[0]] = i;
+                        que.push({dis[now[0]], now[0]});
```

Comment:
```suggestion
                    auto edge = g[v][i];
                    int to = edge[0];
                    long long cap = edge[1], cost = edge[2];
                    if (cap > 0 and dis[to] > dis[v] + cost + h[v] - h[to]) {
                        dis[to] = dis[v] + cost + h[v] - h[to];
                        prevv[to] = v;
                        preve[to] = i;
                        que.push({dis[to], to});
```
`now[0]` や `now[1]` がたくさんあって書くときに混乱しそうなので、もう少し意味のある命名にすると読みやすくなると思います:thumbsup: